src/main/java/org/primefaces/component/cascadeselect/CascadeSelectRenderer.java

```diff
@@ -247,6 +256,36 @@ protected void encodeListItems(FacesContext context, CascadeSelect cascadeSelect
         }
     }
 
+    /**
+     * Recursive method used to find a SelectItem by its value.
+     * @param context FacesContext
+     * @param selectItems the List of SelectItems
+     * @param value the input value to search for
+     * @return either the SelectItem found or NULL if not found
+     */
+    protected SelectItem findItemByValue(FacesContext context, List<SelectItem> selectItems, String value) {
+        SelectItem foundValue = null;
+        for (int i = 0; i < selectItems.size(); i++) {
+            SelectItem item = selectItems.get(i);
+            if (item instanceof SelectItemGroup) {
+                SelectItemGroup selectItemGroup = (SelectItemGroup) item;
+                foundValue = findItemByValue(context, Arrays.asList(selectItemGroup.getSelectItems()), value);
```

Comment:
will Arrays.asList work if selectItemGroup.getSelectItems() is null?

Reply:
Agreed I need to add a NULL guard here because Arrays.asList would throw an NPE if the array is null.